library/std/src/os/uefi/ffi.rs

```diff
@@ -0,0 +1,73 @@
+use crate::ffi::{OsStr, OsString};
+use crate::sealed::Sealed;
+use crate::sys_common::ucs2;
+
+#[unstable(feature = "uefi_std", issue = "none")]
+pub trait OsStrExt: Sealed {
+    /// This function does not do any allocation
+    fn to_ucs2<'a>(&'a self) -> ucs2::EncodeUcs2<'a>;
+
+    /// Creates a UCS-2 Vec which can be passed for FFI.
+    /// Note: The returned string is NULL terminated.
+    /// Note: The supplied should not contain NULL.
```

Comment:
If that is a safety condition, you will need to either assert it, make this function unsafe.

Reply:
Well, the thing is that `NULL` will be transformed into `REPLACEMENT_CHARACTER` by this function. So technically it will produce a valid UEFI string, just different from what the caller might have intended/expected.

Reply:
I have updated the comments to reflect things better. Feel free to reopen this in case that's not enough.